examples/step-42/step-42.cc

```diff
@@ -1553,10 +1559,9 @@ namespace Step42
 
     fraction_of_plastic_q_points_per_cell = 0;
 
-    typename DoFHandler<dim>::active_cell_iterator cell =
-                                                     dof_handler.begin_active(),
-                                                   endc = dof_handler.end();
-    unsigned int cell_number                            = 0;
+    auto         cell        = dof_handler.begin_active();
+    const auto   endc        = dof_handler.end();
+    unsigned int cell_number = 0;
     for (; cell != endc; ++cell, ++cell_number)
```

Comment:
You can convert this into a range-based for loop if you get rid of the `cell_number` variable and instead use `cell->active_cell_index()`.